src/components/Header.tsx

```diff
@@ -0,0 +1,112 @@
+import React, { useContext, useEffect, useRef, useState } from 'react';
+import { TodosContext } from './TodosContext';
+import { TEMPORARY_TODO_ID, USER_ID, addTodo } from '../api/todos';
+import classNames from 'classnames';
+import { Errors } from '../types/Errors';
+import { hideError } from '../functions/hideError';
+
+export const Header: React.FC = () => {
+  const {
+    todos,
+    setTodos,
+    setMessageError,
+    messageError,
+    loadingTodo,
+    setLoadingTodo,
+  } = useContext(TodosContext);
+  const [todoTitle, setTodoTitle] = useState('');
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    inputRef.current?.focus();
+  }, [todos.length, todoTitle, messageError]);
+
+  const AddNewTodo = (
```

Comment:
```suggestion
  const addNewTodo = (
```
name functions in camelCase